agrirouter-api-java-impl/src/test/java/com/dke/data/agrirouter/impl/messaging/SequenceNumberServiceTest.java

```diff
@@ -0,0 +1,63 @@
+package com.dke.data.agrirouter.impl.messaging;
+
+import com.dke.data.agrirouter.api.dto.onboard.OnboardingResponse;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class SequenceNumberServiceTest {
+
+  @Test
+  void givenOnboardingResponseWhenGeneratingSequenceNumberForTheFirstTimeThenTheResultShouldBe1() {
+    OnboardingResponse onboardingResponse = new OnboardingResponse();
+    onboardingResponse.setSensorAlternateId("this-is-my-id");
+    Assertions.assertEquals(
+        1, SequenceNumberService.generateSequenceNumberForEndpoint(onboardingResponse));
+  }
+
+  @Test
+  void
+      givenOnboardingResponseWhenGeneratingSequenceNumberForTwoTimesInARowThenTheResultShouldBe1And2() {
+    OnboardingResponse onboardingResponse = new OnboardingResponse();
+    onboardingResponse.setSensorAlternateId("this-is-another-id");
+    Assertions.assertEquals(
+        1, SequenceNumberService.generateSequenceNumberForEndpoint(onboardingResponse));
+    Assertions.assertEquals(
+        2, SequenceNumberService.generateSequenceNumberForEndpoint(onboardingResponse));
+  }
+
+  @Test
+  void
+      givenTwoOnboardingResponsesWhenGeneratingSequenceNumberForTheFirstTimeThenTheResultShouldBe1ForBoth() {
+    OnboardingResponse firstOnboardingResponse = new OnboardingResponse();
+    firstOnboardingResponse.setSensorAlternateId("this-is-the-first-id");
+    OnboardingResponse secondOnboardingResponse = new OnboardingResponse();
+    secondOnboardingResponse.setSensorAlternateId("this-is-the-second-id");
+    Assertions.assertEquals(
+        1, SequenceNumberService.generateSequenceNumberForEndpoint(firstOnboardingResponse));
+    Assertions.assertEquals(
+        1, SequenceNumberService.generateSequenceNumberForEndpoint(secondOnboardingResponse));
+  }
+
+  @Test
+  void
+      givenTwoOnboardingResponsesWhenGeneratingSequenceNumberForMultipleTimesThenTheResultShouldBeMatchingTheTimesTheMethodWasCalled() {
+    OnboardingResponse firstOnboardingResponse = new OnboardingResponse();
+    firstOnboardingResponse.setSensorAlternateId("this-is-the-first-id-for-multiple-calls");
+    OnboardingResponse secondOnboardingResponse = new OnboardingResponse();
+    secondOnboardingResponse.setSensorAlternateId("this-is-the-second-id-multiple-calls");
+    Assertions.assertEquals(
+        1, SequenceNumberService.generateSequenceNumberForEndpoint(firstOnboardingResponse));
+    Assertions.assertEquals(
+        2, SequenceNumberService.generateSequenceNumberForEndpoint(firstOnboardingResponse));
+    Assertions.assertEquals(
+        3, SequenceNumberService.generateSequenceNumberForEndpoint(firstOnboardingResponse));
+    Assertions.assertEquals(
```

Comment:
Maybe move the last check for firstOnboardingResponse down beyond the last line with secondOnboardingResponse to check (future proof), that the concurrent function for endpoint 1 and endpoint 2 are not mixing up numbers after initialisation?